x/liquidstakeibc/keeper/msg_server.go

```diff
@@ -381,6 +394,95 @@ func (k msgServer) LiquidStake(
 	return &types.MsgLiquidStakeResponse{}, nil
 }
 
+// LiquidStakeLSM defines a method for liquid staking tokens using the LSM
+func (k msgServer) LiquidStakeLSM(
+	goCtx context.Context,
+	msg *types.MsgLiquidStakeLSM,
+) (*types.MsgLiquidStakeLSMResponse, error) {
+	ctx := sdktypes.UnwrapSDKContext(goCtx)
+
+	for _, delegation := range msg.Delegations {
+		// parse the delegator address
+		delegator := sdktypes.MustAccAddressFromBech32(msg.DelegatorAddress)
+
+		// validate the delegation
+		hc, validator, denomTrace, err := k.validateLiquidStakeLSMDeposit(ctx, delegator, delegation)
+		if err != nil {
+			return nil, err
+		}
+
+		// create and store the LSM deposit
+		deposit := &types.LSMDeposit{
+			ChainId:          hc.ChainId,
+			Shares:           sdktypes.NewDecFromInt(delegation.Amount),                                           // TODO: This needs to be updated after https://github.com/iqlusioninc/cosmos-sdk/pull/19
+			Amount:           sdktypes.NewDecFromInt(delegation.Amount).Mul(validator.ExchangeRate).TruncateInt(), // TODO: This needs to be updated after https://github.com/iqlusioninc/cosmos-sdk/pull/19
+			Denom:            denomTrace.BaseDenom,
+			IbcDenom:         delegation.Denom,
+			DelegatorAddress: msg.DelegatorAddress,
+			State:            types.LSMDeposit_DEPOSIT_PENDING,
+			IbcSequenceId:    "",
+		}
+		k.SetLSMDeposit(ctx, deposit)
```

Comment:
we need to set or update, because user can choose to send 50% shares in one txn, next 50% in other.. and the primary key is just chain+delegator+denom